code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -194,6 +194,10 @@ public boolean isComplete() {
     protected Map<String, String> getLabels(DeploymentUnitInstance instance) {
         Service service = instance.getService();
         Map<String, String> labels = new HashMap<>();
+
+        // TODO: This might change in the future if we merge the labels from all services
+        labels.putAll(context.sdService.getServiceLabels(service));
```

Comment:
Ok.  Just took a look.  The changes are safe, but not ideal.  I'll update for better consistency.  Thanks!